zokrates/babyjubjub.py

```diff
@@ -90,3 +96,109 @@ def __mul__(self, n):
 
     def double(self):
         return self.add(self)
+
+    @classmethod
+    def from_x(cls, x):
+        """
+        y^2 = ((a * x^2) / (d * x^2 - 1)) - (1 / (d * x^2 - 1))
+        For every x coordinate, there are two possible points: (x, y) and (x, -y)
+        """
+        assert isinstance(x, FQ)
+        xsq = x * x
+        ax2 = JUBJUB_A * xsq
+        dxsqm1 = inv(JUBJUB_D * xsq - 1, JUBJUB_Q)
+        ysq = dxsqm1 * (ax2 - 1)
+        y = FQ(square_root_mod_prime(int(ysq), JUBJUB_Q))
+        return cls(x, y)
+
+    @classmethod
+    def from_y(cls, y, sign=None):
+        """
+        x^2 = (y^2 - 1) / (d * y^2 - a)
+        """
+        assert isinstance(y, FQ)
+        ysq = y * y
+        lhs = ysq - 1
+        rhs = JUBJUB_D * ysq - JUBJUB_A
+        xsq = lhs / rhs
+        x = FQ(square_root_mod_prime(int(xsq), JUBJUB_Q))
+        if sign is not None:
+            # Used for compress & decompress
+            if (x.n & 1) != sign:
+                x = -x
+        else:
+            if is_negative(x):
+                x = -x
+        return cls(x, y)
+
+    @classmethod
+    def from_hash(cls, entropy):
```

Comment:
Might be worth adding a couple of words to the docstring about the fact that the entropy should be bytes, as I guess people could try to supply an integer here (and it will blow because of the assert, but still would be useful for people checking the docstring).